File: CS/CustomFontEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using System.ComponentModel;
using DevExpress.XtraEditors.Popup;
using System.Drawing;
using System.Windows.Forms;

namespace CustomFontEditControl
{
    class CustomFontEdit : FontEdit
    {
        // static constructor
        static CustomFontEdit() { RepositoryItemCustomFontEdit.RegisterCustomFontEdit(); }

        // public constructor
        public CustomFontEdit() { }

        FontDialogCommonForm protPopupCommonForm;
        FontDialogCommonForm PopupCommonForm
        {
            get
            {
                if (protPopupCommonForm == null)
                {
                    protPopupCommonForm = new FontDialogCommonForm(this.Properties);
                    protPopupCommonForm.FormBorderStyle = FormBorderStyle.FixedDialog;
                    protPopupCommonForm.Text = ""; protPopupCommonForm.ControlBox = false;
                }
                return protPopupCommonForm;
            }        
        }

        // ovverride property
        public override string EditorTypeName
        {
            get { return RepositoryItemCustomFontEdit.CustomFontEditorName; }
        }

        // property as corresponded repositoryitem
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomFontEdit Properties
        {
            get { return base.Properties as RepositoryItemCustomFontEdit; }
        }
        protected override PopupBaseForm CreatePopupForm()
        {
            Rectangle bounds = CalcPopupFormBounds(PopupCommonForm.Size);
            PopupCommonForm.Location = bounds.Location;
            PopupCommonForm.currentEditor = this;
            PopupCommonForm.ShowDialog();
            return null;
        }

        [Browsable(false), EditorBrowsable(EditorBrowsableState.Never)]
        public override int SelectedIndex
        {
            get { return 0; }
            set { }
        }
    }
}

File: VB/CustomFontEdit.vb
Imports DevExpress.XtraEditors
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Popup
Imports System.Drawing
Imports System.Windows.Forms

Namespace CustomFontEditControl

    Friend Class CustomFontEdit
        Inherits FontEdit

        ' static constructor
        Shared Sub New()
            RepositoryItemCustomFontEdit.RegisterCustomFontEdit()
        End Sub

        ' public constructor
        Public Sub New()
        End Sub

        Private protPopupCommonForm As FontDialogCommonForm

        Private ReadOnly Property PopupCommonForm As FontDialogCommonForm
            Get
                If protPopupCommonForm Is Nothing Then
                    protPopupCommonForm = New FontDialogCommonForm(Properties)
                    protPopupCommonForm.FormBorderStyle = FormBorderStyle.FixedDialog
                    protPopupCommonForm.Text = ""
                    protPopupCommonForm.ControlBox = False
                End If

                Return protPopupCommonForm
            End Get
        End Property

        ' ovverride property
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemCustomFontEdit.CustomFontEditorName
            End Get
        End Property

        ' property as corresponded repositoryitem
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemCustomFontEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomFontEdit)
            End Get
        End Property

        Protected Overrides Function CreatePopupForm() As PopupBaseForm
            Dim bounds As Rectangle = CalcPopupFormBounds(PopupCommonForm.Size)
            PopupCommonForm.Location = bounds.Location
            PopupCommonForm.currentEditor = Me
            PopupCommonForm.ShowDialog()
            Return Nothing
        End Function

        <Browsable(False), EditorBrowsable(EditorBrowsableState.Never)>
        Public Overrides Property SelectedIndex As Integer
            Get
                Return 0
            End Get

            Set(ByVal value As Integer)
            End Set
        End Property
    End Class
End Namespace
